app/src/main/java/com/example/simformcafeteria/viewModel/SignupViewModel.kt

```diff
@@ -72,19 +89,18 @@ class SignupViewModel(application: Application): AndroidViewModel(application) {
     }
 
     fun firebaseLoadDepartment() {
-        database.child(getApplication<Application>().resources.getString(R.string.department_node)).addValueEventListener(object : ValueEventListener {
-            override fun onDataChange(dataSnapshot: DataSnapshot) {
-                val tempdepartlistarray = ArrayList<Departments>()
-                for (departmentSnapshot in dataSnapshot.children) {
-                    var dept = Departments(
-                        departmentSnapshot.key?.toInt() ?: 1,
-                        departmentSnapshot.value.toString()
-                    );
-                    tempdepartlistarray.add(dept)
+        database.child(getApplication<Application>().resources.getString(R.string.department_node))
+            .addValueEventListener(object : ValueEventListener {
+                override fun onDataChange(dataSnapshot: DataSnapshot) {
+                    val tempdepartlistarray = ArrayList<Departments>()
```

Comment:
`tempdepartlistarray` is necessary ? can't you add items directly to `departmentList` live data?